cmd/podman/images/scp.go

```diff
@@ -61,16 +68,36 @@ func scp(cmd *cobra.Command, args []string) (finalErr error) {
 		// TODO add tag support for images
 		err error
 	)
+	for i, val := range os.Args {
+		if val == "image" {
+			break
+		} else if i == 0 {
+			continue
+		}
+		parentFlags = append(parentFlags, val)
+	}
+	podman, err := os.Executable()
+	if err != nil {
+		return err
+	}
 	if scpOpts.Save.Quiet { // set quiet for both load and save
 		scpOpts.Load.Quiet = true
 	}
 	f, err := ioutil.TempFile("", "podman") // open temp file for load/save output
 	if err != nil {
 		return err
 	}
-	defer os.Remove(f.Name())
-
 	scpOpts.Save.Output = f.Name()
+	scpOpts.Save.Format = "oci-archive"
+	err = os.Remove(f.Name()) // remove the file and simply use its name so podman creates the file upon save. avoids umask errors
```

Comment:
What if another user creates a file with the same filename after the file has been removed? (theoretically ...)
Is it something to worry about?
If removing the file could be avoided I think it would make for a safer design. 


A side-note: There is an extra "safety belt": _/proc/sys/fs/protected_symlinks_.
I guess most (or all) Linux distributions enable _/proc/sys/fs/protected_symlinks_ 

```
# cat /proc/sys/fs/protected_symlinks
1
# 
```
so that a symlink owned by another user will not be followed.

Quote from the [documentation](https://www.kernel.org/doc/html/latest/admin-guide/sysctl/fs.html#protected-symlinks) :
_When set to “1” symlinks are permitted to be followed only when outside a sticky world-writable directory, or when the uid of the symlink and follower match, or when the directory owner matches the symlink’s owner._


The sticky bit is set.
```
$ stat -c "%a" /var/tmp
1777
$
```

Reply:
In all local save scenarios, I remove the file after it is used. I can't really control the deletion on a remote host though. I used to defer the deletion but since it isn't actually created until the save command is executed, I need to manually do it at the end

Reply:
A second thought. Sorry, I should probably study the code more before commenting about security risks as
I haven't looked into the details.

Reply:
Nit: Use source.File rather then f.Name()